### ISSUE TITLE
Tactic name of "}" not rendered correctly or indexed in sphinx.

### ISSUE THREAD

**Issue by jfehrle** [coq/coq#13623]
```
.. tacn:: %}
   :name: %}
```

in Sphinx is rendered as

![image](https://user-images.githubusercontent.com/1253341/101976548-0dd7ab00-3bfb-11eb-9979-8ac790253cce.png)

I've tried many permutations.  The percent signs are necessary in order for it to parse.  Not sure whether to look in `TacticNotations.g`, `coqdomain.py` or in Sphinx.  Somehow it's getting split into multiple tacns.

@cpitclaudel, do you have any suggestions?

**Comment by cpitclaudel** [comment]
An indentation issue? I can reproduce your issue if I put 4 spaces instead of 3, or a tab, but the code you posted works fine.
Also, I don't think you need the second `%` sign.

**Comment by jfehrle** [comment]
Could it be a difference in Sphinx versions?  What version do you have?
IIRC I got an error if leave out t either %.

**Comment by cpitclaudel** [comment]
```
$ sphinx-build --version
sphinx-build 3.0.4
```

Please show the error?

**Comment by jfehrle** [comment]
I have 2.3.1, which CI also uses, perhaps we could go to the newer version?  Maybe `./configure` should check the Sphinx version, too?  And Python and other things as well?

Unless we choose to go to the latest version in CI, I'm reluctant to run `apt upgrade` because I don't know how to reverse it.  `apt list | grep sphinx` shows about 50 packages, none of them with a version containing `2.3.1`.  `apt update | grep sphinx` shows nothing.

```
.. tacn:: %}
   :name: }

Extension error:
proof-mode.rst:477 Parse error in notation!
Offending notation: :name: }
Error message: 1:6: mismatched input ' ' expecting <EOF>
```

```
.. tacn:: }
   :name: %}

line 1:0 mismatched input '}' expecting {'{|', LGROUP, '{', ESCAPED, '|', ATOM, ID}

Extension error:
proof-mode.rst:477 Parse error in notation!
Offending notation: }
Error message: 1:0: mismatched input '}' expecting {'{|', LGROUP, '{', ESCAPED, '|', ATOM, ID}
```

**Comment by cpitclaudel** [comment]
> Unless we choose to go to the latest version in CI, I'm reluctant to run apt upgrade because I don't know how to reverse it.

Sphinx usually comes from pip, not apt, so reversing would be `pip3 install sphinx==2.3.1`.  I did that right now, and I still can't reproduce the issue:

![image](https://user-images.githubusercontent.com/8181630/101999042-d89d8c80-3cd0-11eb-968d-8d00f8145513.png)

![image](https://user-images.githubusercontent.com/8181630/101999061-fe2a9600-3cd0-11eb-8ac4-ea0b6f8f9bb3.png)

![image](https://user-images.githubusercontent.com/8181630/101999052-ee12b680-3cd0-11eb-9509-cdaa380e11d6.png)

Are you using a clean clone or do you have other local changes?

**Comment by jfehrle** [comment]
You tried left curly brace.  The problem is with right curly brace.

**Comment by cpitclaudel** [comment]
With Sphinx 2.3.1 I get this:

![image](https://user-images.githubusercontent.com/8181630/102003194-f8967580-3cfb-11eb-8f9d-fd063721a98e.png)
![image](https://user-images.githubusercontent.com/8181630/102003183-e1578800-3cfb-11eb-8c9f-2741fa15ec85.png)
![image](https://user-images.githubusercontent.com/8181630/102003191-efa5a400-3cfb-11eb-8321-9cbb1251a7a9.png)

**Comment by cpitclaudel** [comment]
```
00000000  2e 2e 20 74 61 63 6e 3a  3a 20 25 7d 0a 20 20 20  |.. tacn:: %}.   |
00000010  3a 6e 61 6d 65 3a 20 7d  0a 0a 20 20 20 58 59 5a  |:name: }..   XYZ|
00000020  2e 0a                                             |..|
00000022
```

**Comment by cpitclaudel** [comment]
The weird thing here is that this is an issue happening at the level of the parser: somehow it's determining that `:name:` isn't an option and so it's parsing it as a second "signature".

**Comment by jfehrle** [comment]
It fails the same way when I install sphinx 3.3.1.  My branch doesn't change any python code, only rsts and doc_grammar-related files.  And it fails in CI with 2.3.1, see `doc:refman` in https://github.com/coq/coq/pull/13629

```
.. tacn:: %}
   :name: }

   xyz

line 1:6 mismatched input ' ' expecting <EOF>

Extension error:
proof-mode.rst:468 Parse error in notation!
Offending notation: :name: }
Error message: 1:6: mismatched input ' ' expecting <EOF>
```

**Comment by cpitclaudel** [comment]
> So it looks like your setup is the unusual one--do you have any modifications?

Huh? No, I'm using a fresh clone from master.

I presume the issue is the file it appears in, which would be in line with a parser issue.

**Comment by jfehrle** [comment]
The problem was 99 lines after the line reported in the error message; the `seealso` should have 3 spaces of indentation, not 2.  I thought Sphinx would complain about this sort of thing. Ugh!

```
.. tacn:: %}
   :name: }

:

  .. seealso:: The error messages for bullets below.
```

**Comment by jfehrle** [comment]
However, the downgrade to 2.3.1 doesn't work for me.  For now, I'll just use the newer version.

```
$ pip3 install sphinx==2.3.1

:

$ make sphinx-clean;time make -j 12 sphinx SPHINXWARNERROR=0

:

Running Sphinx v2.3.1

Exception occurred:
  File "/home/jfehrle/.local/lib/python3.6/site-packages/sphinx/builders/html/__init__.py", line 1247, in setup
    app.connect('config-inited', convert_html_css_files, priority=800)
TypeError: connect() got an unexpected keyword argument 'priority'
The full traceback has been saved in /tmp/sphinx-err-1cqq5guf.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
```

**Comment by jfehrle** [comment]
One other question: do you know if there is an easy way to get, say, the tactics index to be in 2 columns like the general index?

**Comment by cpitclaudel** [comment]
>  I thought Sphinx would complain about this sort of thing.

It's a case of being too forgiving: it determines what indentation the body has by looking at the least indented line :/

> One other question: do you know if there is an easy way to get, say, the tactics index to be in 2 columns like the general index?

It's in the theme files; there are different style for domain indices and the general index.

**Comment by jfehrle** [comment]
Do you mean in `theme.css`? ?

I see there are classes `genindextabnle` and `modindextable` but I don't see how either one is specified.  Changing the classname in the browser debugger doesn't have any effect.

**Comment by cpitclaudel** [comment]
No no, I mean the html templates.  I'm not super sure how this works but I'm guessing it's in these files: [https://github.com/sphinx-doc/sphinx/blob/3.x/sphinx/themes/basic/genindex.html](general index template), [https://github.com/sphinx-doc/sphinx/blob/3.x/sphinx/themes/basic/domainindex.html](domain index template); you can see them being referred to in the html generation code: https://www.sphinx-doc.org/en/master/_modules/sphinx/builders/html.html

**Comment by jfehrle** [comment]
I'll look at them.  To override them, I copy them to our source tree and/or update conf.py?

**Comment by cpitclaudel** [comment]
See https://www.sphinx-doc.org/en/master/templating.html , specifically this:

> To customize the output of your documentation you can override all the templates (both the layout templates and the child templates) by adding files with the same name as the original filename into the template directory of the structure the Sphinx quickstart generated for you.

There's an example in the `_templates` directory, which I did to get the version switcher.

**Comment by Zimmi48** [comment]
@jfehrle Can we close this?

**Comment by jfehrle** [comment]
It appears to be non-trivial.  I experimented with both `domainindex.html` and `genindex.html`.  https://www.sphinx-doc.org/en/master/_modules/sphinx/builders/html.html seems to have some special handling to even up the columns.  I don't see how the generated html for the general index gets 2 columns, either.  Perhaps I'll try again some other time.

I would also want to make only some of the domain indexes 2 columns.  Most of the error messages would be split across 2 lines, which seems undesirable.  When I set the column width to 50%, then a few items in the options index get split.  I guess that's OK.

![image](https://user-images.githubusercontent.com/1253341/102726323-b7d9c600-42d2-11eb-984b-e511a73ac0a1.png)
